src/path.rs

```diff
@@ -0,0 +1,260 @@
+use std::borrow::Cow;
+
+use serde_json::json;
+
+use crate::json::JsonLike;
+use crate::lambda::{EvaluationContext, ResolverContextLike};
+
+///
+/// The path module provides a trait for accessing values from a JSON-like structure.
+///
+
+///
+/// The PathString trait provides a method for accessing values from a JSON-like structure.
+/// The returned value is encoded as a plain string.
+/// This is typically used in evaluating mustache templates.
+///
+pub trait PathString {
+  fn path_string<T: AsRef<str>>(&self, path: &[T]) -> Option<Cow<'_, str>>;
+}
+
+///
+/// The PathGraphql trait provides a method for accessing values from a JSON-like structure.
+/// The returned value is encoded as a GraphQL Value.
+///
+pub trait PathGraphql {
+  fn path_graphql<T: AsRef<str>>(&self, path: &[T]) -> Option<String>;
+}
+
+impl PathString for serde_json::Value {
+  fn path_string<T: AsRef<str>>(&self, path: &[T]) -> Option<Cow<'_, str>> {
+    self.get_path(path).and_then(|a| match a {
+      serde_json::Value::String(s) => Some(Cow::Borrowed(s.as_str())),
+      serde_json::Value::Number(n) => Some(Cow::Owned(n.to_string())),
+      serde_json::Value::Bool(b) => Some(Cow::Owned(b.to_string())),
+      _ => None,
+    })
+  }
+}
+
+impl PathGraphql for serde_json::Value {
+  fn path_graphql<T: AsRef<str>>(&self, path: &[T]) -> Option<String> {
+    self
+      .get_path(path)
+      .and_then(|v| async_graphql::Value::from_json(v.clone()).ok().map(|v| v.to_string()))
```

Comment:
As a separate note: This is going to be quite slow.

Reply:
I've removed this particular code as I mentioned in discord it doesn't bring much value